media/public/include/IControl.h

```diff
@@ -81,14 +86,18 @@ class IControl
     IControl &operator=(IControl &&) = delete;
 
     /**
-     * @brief Register new IControlClient
+     * @brief Register new IControlClient. This method will hold a shared_ptr
+     *        to the client until the destruction of the IControl object.
+     *        At destruction the client will be unregistered and the
+     *        shared_ptr will be released
      *
      * @param[in]  client   : Client object for callbacks
      * @param[out] appState : Current application state
      *
      * @retval true on success, false otherwise.
      */
-    virtual bool registerClient(std::weak_ptr<IControlClient> client, ApplicationState &appState) = 0;
+    virtual bool registerClientAndUnregisterOnDestruction(std::weak_ptr<IControlClient> client,
```

Comment:
If you want to change method name only, I'd keep the old one to skip public interface change

Reply:
But the old name didn't say what he method actually does...

Reply:
ok, so let's wait for Adam and Luke opinion about that ;)

Reply:
Name is reverted back to registerClient and a comment is used to highlight the difference between IClientController::registerClient and IControl::registerClient